pkg/commands/drift/statefiles/statefiles.go

```diff
@@ -0,0 +1,214 @@
+func (c *DriftStatefilesCommand) Process(ctx context.Context) error {
+	logger := logging.FromContext(ctx).
+		Named("drift.statefiles").
+		With("github_owner", c.GitHubFlags.FlagGitHubOwner).
+		With("github_repo", c.GitHubFlags.FlagGitHubOwner)
+
+	logger.Debug("starting Guardian drift statefiles")
+	logger.Debug("finding entrypoint directories")
+
+	entrypoints, err := terraform.GetEntrypointDirectories(c.directory)
+	if err != nil {
+		return fmt.Errorf("failed to find terraform directories: %w", err)
+	}
+	entrypointBackendFiles := make([]string, 0, len(entrypoints))
+	for _, e := range entrypoints {
+		entrypointBackendFiles = append(entrypointBackendFiles, e.BackendFile)
+	}
+	logger.Debugw("terraform entrypoint directories", "entrypoint_backend_files", entrypointBackendFiles)
+
+	expectedURIs := make([]string, 0, len(entrypointBackendFiles))
+	gcsBuckets := make([]string, 0, len(entrypointBackendFiles))
+	var errs []error
+	for _, f := range entrypointBackendFiles {
+		config, _, err := terraform.ExtractBackendConfig(f)
+		if err != nil {
+			errs = append(errs, fmt.Errorf("failed to parse Terraform backend config: %w", err))
+			continue
+		}
+		if config.GCSBucket == nil || *config.GCSBucket == "" {
+			errs = append(errs, fmt.Errorf("unsupported backend type for terraform config at %s - only gcs backends are supported", f))
+			continue
+		}
+		gcsBuckets = append(gcsBuckets, *config.GCSBucket)
+		expectedURIs = append(expectedURIs, fmt.Sprintf("gs://%s/%s/default.tfstate", *config.GCSBucket, *config.Prefix))
+	}
+	if len(errs) > 0 {
+		return fmt.Errorf("failed to determine statefile gcs URIs: %w", errors.Join(errs...))
+	}
+
+	gcsBucket := c.flagGCSBucket
+	for _, bucket := range gcsBuckets {
+		if gcsBucket == "" {
+			gcsBucket = bucket
+		}
+		if bucket != gcsBucket {
+			return fmt.Errorf("found multiple definitions for gcs buckets - expected a single gcs bucket; from configs: %v; expected %s", gcsBuckets, gcsBucket)
+		}
+	}
+
+	if gcsBucket == "" {
+		return fmt.Errorf("unable to determine gcs bucket - please provide the gcsBucket flag or point at a terraform config with gcs backends")
+	}
+
+	logger.Debug("finding statefiles in gcs bucket", "gcs_bucket")
+
+	gotURIs, err := c.terraformParser.StateFileURIs(ctx, []string{gcsBucket})
+	if err != nil {
+		return fmt.Errorf("failed to determine state file URIs for gcs bucket %s: %w", gcsBucket, err)
+	}
+
+	statefilesNotInRemote := sets.SubtractMapKeys(Set(expectedURIs), Set(gotURIs))
+	statefilesNotInLocal := sets.SubtractMapKeys(Set(gotURIs), Set(expectedURIs))
+
+	changesDetected := len(statefilesNotInRemote) > 0 || len(statefilesNotInLocal) > 0
+	m := driftMessage(statefilesNotInRemote, statefilesNotInLocal)
+	if changesDetected {
+		c.Outf(m)
+	}
+
+	return nil
+}
+
+func Set(values []string) map[string]struct{} {
+	set := make(map[string]struct{})
+	for _, v := range values {
+		set[v] = struct{}{}
+	}
+	return set
+}
```

Comment:
I'm only seeing this used above on L178-179 where we're converting slices to maps. Would it be better to subtract the slices directly and drop this function?

```go
	statefilesNotInRemote := sets.Subtract(expectedURIs, gotURIs)
	statefilesNotInLocal := sets.Subtract(gotURIs, expectedURIs)
```

Then you could also avoid `maps.Keys` and `sort.Strings` on L201-202 and L209-210.

Reply:
I still want the sort so that the output to the user is in a reliable order. Updated the rest